common/txmgr/types/models.go

```diff
@@ -0,0 +1,16 @@
+package types
+
+type TxmErrorType int
+
+// Generalized transaction manager error types that dictates what should be the next action, depending on the RPC error response.
+const (
+	Successful TxmErrorType = iota
+	Fatal
```

Comment:
Can you also please add comments to explain each error type, and when and why will they be returned?
Ideally someone reading broadcaster code should understand how to handle each error here, without going into client implementation code to find how it works.